Makefile

```diff
@@ -30,28 +41,41 @@ KOTS_VERSION = v$(shell echo $(ADMIN_CONSOLE_CHART_VERSION) | sed 's/\([0-9]\+\.
-	-X github.com/replicatedhq/embedded-cluster/pkg/addons/openebs.UtilsVersion=$(OPENEBS_UTILS_VERSION) \
-	-X github.com/replicatedhq/embedded-cluster/pkg/addons/seaweedfs.Version=$(SEAWEEDFS_CHART_VERSION) \
-	-X github.com/replicatedhq/embedded-cluster/pkg/addons/registry.Version=$(REGISTRY_CHART_VERSION) \
-	-X github.com/replicatedhq/embedded-cluster/pkg/addons/registry.ImageVersion=$(REGISTRY_IMAGE_VERSION) \
-	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.Version=$(VELERO_CHART_VERSION) \
-	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.VeleroTag=$(VELERO_IMAGE_VERSION) \
-	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.AwsPluginTag=$(VELERO_AWS_PLUGIN_IMAGE_VERSION)
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/embeddedclusteroperator.EmbeddedOperatorChartVersion=$(EMBEDDED_OPERATOR_CHART_VERSION) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/embeddedclusteroperator.EmbeddedOperatorImageTag=$(EMBEDDED_OPERATOR_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/embeddedclusteroperator.EmbeddedOperatorProxyImageTag=$(EMBEDDED_OPERATOR_PROXY_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/embeddedclusteroperator.EmbeddedOperatorUtilsImageTag=$(EMBEDDED_OPERATOR_UTILS_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/embeddedclusteroperator.EmbeddedOperatorImageOverride=$(EMBEDDED_OPERATOR_IMAGE_OVERRIDE) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/openebs.OpenEBSChartVersion=$(OPENEBS_CHART_VERSION) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/openebs.OpenEBSImageTag=$(OPENEBS_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/openebs.OpenEBSUtilsImageTag=$(OPENEBS_UTILS_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/openebs.OpenEBSKubectlImageTag=$(OPENEBS_KUBECTL_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/seaweedfs.SeaweedFSChartVersion=$(SEAWEEDFS_CHART_VERSION) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/seaweedfs.SeaweedFSImageTag=$(SEAWEEDFS_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/registry.RegistryChartVersion=$(REGISTRY_CHART_VERSION) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/registry.RegistryImageTag=$(REGISTRY_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.VeleroChartVersion=$(VELERO_CHART_VERSION) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.VeleroImageTag=$(VELERO_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.VeleroAWSPluginImageTag=$(VELERO_AWS_PLUGIN_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.VeleroKubectlImageTag=$(VELERO_KUBECTL_IMAGE_TAG) \
+	-X github.com/replicatedhq/embedded-cluster/pkg/addons/velero.VeleroRestoreHelperImageTag=$(VELERO_RESTORE_HELPER_IMAGE_TAG)
```

Comment:
I am inclined to make the automation to write this down to a `json` file in each add-on directory that then can be embedded. What do you think ? We would not need to have any of these tags.

Reply:
along the same lines it would be pretty cool since buildtools discovers images in charts if it were to generate the values yaml file in each add-on directory to override images